batcher/aligned-batcher/src/lib.rs

```diff
@@ -84,6 +94,84 @@ impl BatchState {
             .and_modify(|count| *count += 1)
             .or_insert(1);
     }
+
+    fn get_entry(&self, sender: Address, nonce: U256) -> Option<&BatchQueueEntry> {
+        self.batch_queue
+            .iter()
+            .map(|(entry, _)| entry)
+            .find(|entry| {
+                entry.sender == sender
+                    && U256::from_big_endian(entry.nonced_verification_data.nonce.as_slice())
+                        == nonce
+            })
+    }
+
+    fn check_validity_and_increment_fee(
+        &mut self,
+        entry: BatchQueueEntry,
+    ) -> Option<ValidityResponseMessage> {
+        let max_fee = entry.nonced_verification_data.max_fee;
+        let nonce = U256::from_big_endian(entry.nonced_verification_data.nonce.as_slice());
+        let sender = entry.sender;
+
+        debug!(
+            "Checking validity of entry with sender: {:?}, nonce: {:?}, max_fee: {:?}",
+            sender, nonce, max_fee
+        );
+
+        // it is a valid entry only if there is no entry with the same sender, lower nonce and a lower fee
+        let is_valid = !self.batch_queue.iter().any(|(entry, _)| {
+            entry.sender == sender
+                && U256::from_big_endian(entry.nonced_verification_data.nonce.as_slice()) < nonce
+                && entry.nonced_verification_data.max_fee < max_fee
+        });
+
+        if !is_valid {
+            return Some(ValidityResponseMessage::InvalidMaxFee);
+        }
+
+        info!(
+            "Entry is valid, incrementing fee for sender: {:?}, nonce: {:?}, max_fee: {:?}",
+            sender, nonce, max_fee
+        );
+
+        // remove the old entry and insert the new one
+        self.batch_queue.remove(&entry);
+        self.batch_queue
+            .push(entry.clone(), BatchQueueEntryPriority::new(max_fee, nonce));
```

Comment:
I guess this is something related to the `PriorityQueue` API, but it seems a little confusing that the old and new entries are the same. Some comment will be useful here

Reply:
changed on d49737fcbc8e696e47faf238262103d81c45410b